### ISSUE TITLE
IsValidModel should be changed

### ISSUE THREAD

**Issue by Allerek** [multitheftauto/mtasa-blue#1849]
**Version**
nightly 1.5.8-20716 onwards

**Additional context**
#IMPORANT: I'm C++ Amateur, what im saying can be a complete CRAP!#

For now IsValidModel in *Client\mods\deathmatch\logic\CVehicleNames.cpp* returns true only if we check for standard GTA:SA models.
*ulModel >= 400 && ulModel <= 611;* 

We should change it to use 

getModelInfo(ulModel)->isVehicle();

in order to return real values when using vehicles from engineRequestModel.

This will allow us to add functions from #1175

**Comment by Allerek** [comment]
^^Not a bug if thinking more, but can cause problems in future versions

**Comment by Pirulax** [comment]
Seems reasonable.

**Comment by TheNormalnij** [comment]
We need good vehicle manager with dynamic id support first

**Comment by Pirulax** [comment]
I think all these `CClientVehicle::IsValidModel` and such functions should be moved to `CClientModelManager`, so for example.: `IsVehicleModel` would be an alias for `GetModelType() == eClientModelType::VEHICLE`, same for peds, and objects.

**Comment by TheNormalnij** [comment]
`CModelInfoSA` is valid model manager. `CClientModelManager` is only manager for allocated ID's

**Comment by Pirulax** [comment]
Yes I know, but I'd like to use CCLientModelManger as we'll have to support custom model names as well.

**Comment by FernandoMTA** [comment]
I think this should be prioritized more. Vehicle add-ons are a game-changing feature.

**Comment by Fernando-A-Rocha** [comment]
@BCG2000 sorry for the tag but do you think this still needs to be resolved?

**Comment by BCG2000** [comment]
This can only be fixed when https://github.com/multitheftauto/mtasa-blue/issues/2251 is implemented, as revamping functions like this one is part of implementing it. Until then, modifying these functions individually should be avoided, since that is only working around the root issue instead of directly addressing it.

**Comment by Fernando-A-Rocha** [comment]
> This can only be fixed when https://github.com/multitheftauto/mtasa-blue/issues/2251 is implemented, as revamping functions like this one is part of implementing it. Until then, modifying these functions individually should be avoided, since that is only working around the root issue instead of directly addressing it.

Just as I thought! I have the same opinion. I think we have enough temporary solutions for new vehicle models to work. We'll get the best of it once server side allocation is a thing.
